fdbserver/ApplyMetadataMutation.cpp

```diff
@@ -297,6 +297,35 @@ class ApplyMetadataMutationsImpl {
 		}
 	}
 
+	void checkSetStorageServerToTeamIDKeyPrefix(MutationRef m) {
+		if (!m.param1.startsWith(storageServerToTeamIdKeyPrefix) || !SERVER_KNOBS->ENABLE_PARTITIONED_TRANSACTIONS) {
+			return;
+		}
+		ASSERT_WE_THINK(tLogGroupCollection.isValid());
+		UID serverID = decodeStorageServerToTeamIdKey(m.param1);
+		ptxn::StorageServerStorageTeams teams(
+		    m.param2); // assuming it is full set, could it be true that it only contain incremental data?
+
+		auto it = ssToStorageTeam->find(serverID);
+		if (it != ssToStorageTeam->end()) {
+			TraceEvent("ReplaceSSTeams", dbgid)
+			    .detail("SS", serverID)
+			    .detail("Teams", describe(teams.getStorageTeams()));
+			ssToStorageTeam->erase(it);
+		} else {
+			TraceEvent("AssignSSTeams", dbgid)
+			    .detail("SS", serverID)
+			    .detail("Teams", describe(teams.getStorageTeams()));
+		}
+		ssToStorageTeam->emplace(serverID, teams);
```

Comment:
I think this is the correct place to set `ssToStorageTeam`, which means in below `checkSetStorageTeamIDKeyPrefix`, it shouldn't change it. Can you remove populating `ssToStorageTeam` in `checkSetStorageTeamIDKeyPrefix`?